[RDY]Move hardcoded temperature effects to functions

Coolthulhu opened the pull request:
Effects hot, cold and frostbite moved to a function.
Also cleaned them up: instead of those big switches, I changed it to `map` keyed by `body_part, intensity` and having `struct`s with all the effects. Saves a lot of lines and improves readability.

Also fixed effect miss messages not being shown.

DangerNoodle commented:
This will be interesting to see.

BrettDong commented:
There is already an issue (#20829) reporting 'Changing in-game language not affect some strings'. Initializing const containers with gettext() returned strings is adding to that issue. If the player changes language, those strings will not be updated to the new language. Could you leave the strings in const maps in their original English form, and call gettext() when displaying the message?

Coolthulhu commented:
OK, fixed the static translations. Changed it to store untranslated string and translate each time.

Also fixed an error (present in master) where miss messages caused by effects would be cleared before they could be shown to the player.

Coolthulhu commented:
Jenkins, rebuild

Coolthulhu commented:
I'll need to add the translate extraction macro/function to it.

Coolthulhu commented:
All right, added the translate macros. This should be ready now.

codemime commented on src/player_hardcoded_effects.cpp:
```diff
@@ -273,6 +273,137 @@ static void eff_fun_hallu( player &u, effect &it )
+static void eff_fun_cold( player &u, effect &it )
+{
+    // { body_part, intensity }, { str_pen, dex_pen, int_pen, per_pen, msg, msg_chance, miss_msg }
+    const std::map<std::pair<body_part, int>, temperature_effect> effs = {{
```
This probably doesn't matter much since compilers would likely optimize the assignment anyways, but this is a good candidate for being `static`.
